android/app/src/main/java/org/fossasia/openevent/viewmodels/SearchActivityViewModel.java

```diff
@@ -0,0 +1,112 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.text.TextUtils;
+
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Track;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import io.realm.Case;
+import io.realm.Realm;
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class SearchActivityViewModel extends ViewModel {
+
+    private String searchText = "";
+    private List<Object> objectList = new ArrayList<>();
+    private Realm realm;
+    private MutableLiveData<List<Object>> searchResults = new MutableLiveData<>();
+
+    public SearchActivityViewModel() {
+        realm = Realm.getDefaultInstance();
+    }
+
+    public LiveData<List<Object>> getSearchResults(String search) {
+        objectList.clear();
+        String wildcardQuery = String.format("*%s*", search.toLowerCase(Locale.getDefault()));
+        searchText = search;
+        if (!TextUtils.isEmpty(wildcardQuery)) {
```

Comment:
If you look closely, wildcardQuery can NEVER be empty. It will always at least be ``**'`. So you need this check before formatting, not after